client/src/app/dashboard/table-all/table-all.component.ts

```diff
@@ -16,53 +16,160 @@
  * limitations under the License.
  */
 
-import { Component, OnInit, Input } from '@angular/core';
-import {Instance} from '../../api';
-import { SelectionModel} from '@angular/cdk/collections';
-import { MatDialog, MatTableDataSource} from '@angular/material';
+import { Component, OnInit, Input, ViewChild, ɵstringify } from '@angular/core';
+import { Instance } from '../../api';
+import { MatDialog, MatDialogRef, MAT_DIALOG_DATA, MatTable, MatPaginator, MatTableDataSource } from '@angular/material';
+import { DeleteDialogComponent } from '../delete-dialog/delete-dialog.component';
+import { AddDialogComponent } from '../add-dialog/add-dialog.component';
+import { ApiService } from '../../api';
+
 
 
 @Component({
-  selector: 'app-table-all',
-  templateUrl: './table-all.component.html',
-  styleUrls: ['./table-all.component.css']
+    selector: 'app-table-all',
+    templateUrl: './table-all.component.html',
+    styleUrls: ['./table-all.component.css']
 })
 export class TableAllComponent implements OnInit {
+    @Input() type: Instance.ComponentTypeEnum;
 
-  @Input() set data_array(data_array: Instance[]) {
-    if (this.dataSource != null) {
-      this.dataSource = new MatTableDataSource<Instance>(data_array);
-    } else {
-      this.dataSource.data = data_array;
+    @Input() set data_array(data_array: Instance[]) {
+        if (this.dataSource != null) {
+            this.dataSource = new MatTableDataSource<Instance>(data_array);
+        } else {
+            this.dataSource.data = data_array;
+        }
     }
-  }
-  displayedColumns = ['ID', 'name', 'host', 'portNumber', 'select'];
-  dataSource: MatTableDataSource<Instance> = new MatTableDataSource<Instance>(this.data_array);
-  selection = new SelectionModel<Instance>(true, []);
-  dialogResult: any;
+    @Input() compType: string;
 
-  constructor(public dialog: MatDialog) {
 
-  }
+    displayedColumns = ['ID', 'name', 'host', 'portNumber', 'instanceState', 'action'];
+    dataSource: MatTableDataSource<Instance> = new MatTableDataSource<Instance>(this.data_array);
+    dialogResult: any;
+    @ViewChild(MatTable) table: MatTable<any>;
```

Comment:
please don't use the any data type.

Reply:
this ist still not resolved in the new version.

Reply:
@windor12  is currently in progress adding her test cases and changing the remaining data types.